src/metabase/models/user.clj

```diff
@@ -188,13 +188,23 @@
            common-name)
       (assoc :common_name common-name))))
 
+(defn- maybe-get-name-from-api-key
+  "If the user is an API Key user, replace any existing name with the name from the API Key table."
+  [{:keys [type] :as user}]
+  (cond-> user
+    (= :api-key type)
+    (assoc :first_name (i18n/tru "API Key:")
+           :last_name (:name (t2/select-one [:model/ApiKey :name] :user_id (:id user))))))
```

Comment:
nit: marginally more concise (I think this is the syntax)
```suggestion
           :last_name (t2/select-one-fn :name :model/ApiKey :user_id (:id user)))))
```

Reply:
also maybe worth adding some error handling here in case the `api-key` user doesn't have an associated `ApiKey` model? shouldn't happen in practice but I don't think we enforce that as a DB invariant

Reply:
I tried to do this and it ended up breaking API key creation, because we need to:
- insert a new `:api-key`-typed user, grabbing the ID
- insert a new ApiKey with that `user_id`
Adding error-handling here causes this to break, because `t2/insert-returning-pk!` or `t2/insert-returning-instance!` both run after-select hooks.

Maybe we could set the `:last_name` to `(tru "[Missing]")` or something along those lines? Not ideal, but I'm not sure of a better option...

Reply:
Oh I see. I guess a workaround could be to avoid those functions when inserting the user and just use `t2/insert!`, and then `t2/query` to fetch the ID. But I don't think this error handling is _vital_, it's just nice to have.

cc @camsaul if there's a better t2 pattern to use in this sort of case; I don't really know the internals.